integrationtests/src/test/java/org/neo4j/causalclustering/scenarios/ListQueriesProcedureInClusterIT.java

```diff
@@ -0,0 +1,171 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.scenarios;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.neo4j.causalclustering.core.CoreGraphDatabase;
+import org.neo4j.causalclustering.discovery.Cluster;
+import org.neo4j.causalclustering.discovery.CoreClusterMember;
+import org.neo4j.causalclustering.readreplica.ReadReplicaGraphDatabase;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.impl.factory.GraphDatabaseFacade;
+import org.neo4j.test.causalclustering.ClusterRule;
+import org.neo4j.test.rule.VerboseTimeout;
+import org.neo4j.test.rule.concurrent.ThreadingRule;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+
+public class ListQueriesProcedureInClusterIT {
+    private final ClusterRule clusterRule =
+            new ClusterRule(getClass()).withNumberOfCoreMembers(3).withNumberOfReadReplicas(1);
+    private final VerboseTimeout timeout = VerboseTimeout.builder()
+            .withTimeout(1000, TimeUnit.SECONDS)
+            .build();
+
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule(clusterRule).around(timeout);
+
+    private Cluster cluster;
+
+    @Rule
+    public final ThreadingRule threads = new ThreadingRule();
+
+    @Before
+    public void setup() throws Exception {
+        cluster = clusterRule.startCluster();
+    }
+
+    @Test
+    public void listQueriesWillNotIncludeQueriesFromOtherServersInCluster() throws Exception {
+        // given
+        String CORE_QUERY = "MATCH (n) SET n.number = n.number - 1";
+        String REPLICA_QUERY = "MATCH (n) RETURN n";
+        CountDownLatch resourceLocked = new CountDownLatch(1);
+        CountDownLatch listQueriesLatchOnCore = new CountDownLatch(1);
+        CountDownLatch executedCoreQueryLatch = new CountDownLatch(1);
+        ReadReplicaGraphDatabase replicaDb = cluster.findAnyReadReplica().database();
+
+        final Node[] node = new Node[1];
+
+        //Given
+        CoreClusterMember leader = cluster.coreTx((CoreGraphDatabase leaderDb, Transaction tx) -> {
+            node[0] = leaderDb.createNode();
+            tx.success();
+            tx.close();
+        });
+
+        CoreGraphDatabase leaderDb = leader.database();
+        Result matchAllResult = leaderDb.execute("MATCH (n) RETURN n");
+        assertTrue(matchAllResult.hasNext());
+
```

Comment:
Add `matchAllResult.close()` here to make sure that the transaction implied by the query gets closed properly.